### PR TITLE
F/os 738 move commons to a single package

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task ID: [OS-738]()

## Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them when possible.
- [ ] I have updated the `CHANGELOG.md` file in the root folder of the package after the `[UPCOMING]` title and before the latest version.
- [ ] I have tested my code on the test network.

[OS-738]: https://aragonassociation.atlassian.net/browse/OS-738?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Rekard0 — DISMISSED** [review]

File: modules/client-common/src/constants.ts [hunk 1]
```diff
@@ -407,3 +408,13 @@ Object.freeze(Permissions);
 export { Permissions };
 Object.freeze(PermissionIds);
 export { PermissionIds };
+export const IPFS_CID_REGEX =
```
Comment (heueristik) [inline modules/client-common/src/constants.ts:411]: The regex only applies to CIDv0. Note, that there is also CIDv1 (https://ipfs-search.readthedocs.io/en/latest/ipfs_datatypes.html#cid)

File: modules/client-common/src/constants.ts [hunk 2]
```diff
@@ -407,3 +408,13 @@ Object.freeze(Permissions);
...
+  /^Qm([1-9A-HJ-NP-Za-km-z]{44,}|b[A-Za-z2-7]{58,}|B[A-Z2-7]{58,}|z[1-9A-HJ-NP-Za-km-z]{48,}|F[0-9A-F]{50,})$/;
+
+export const IPFS_URI_REGEX =
```
Comment (heueristik) [inline modules/client-common/src/constants.ts:414]: See above

File: modules/client-common/src/encoding.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+ * @return {*}  {{ pluginAddress: string; id: number }}
+ */
+export function decodeProposalId(
+  proposalId: string,
+): { pluginAddress: string; id: number } {
+  if (!isProposalId(proposalId)) {
+    throw new InvalidProposalIdError();
+  }
+  // const matchedRegexResult =
+  //   proposalId.match(/^(0x[A-Fa-f0-9]{40})_(0x[A-Fa-f0-9]{1,64})$/) || [];
```
Comment (heueristik) [inline modules/client-common/src/encoding.ts]: ?

File: modules/client-common/test/unit/encoding.test.ts [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+      expect(() => decodeRatio(10 ** 10, 9)).toThrow(
+        "The value is out of range",
+      );
+    });
+  });
+  describe("encodeProposalId", () => {
+    it("Should encode a nonce-based proposalId", () => {
+      const entries = [
+        {
+          addr: "0x0000000000000000000000000000000000000000",
```
Comment (heueristik) [inline modules/client-common/test/unit/encoding.test.ts:212]: In OSx, we use format string for better readability. E.g.,
```ts
`0x${'0'.repeat(39)}1`, // address(1)
```
Maybe you want to use this too

File: modules/client-common/src/constants.ts (hunk 1 above)
Comment (josemarinas) [reply modules/client-common/src/constants.ts:411]: I think is missing some parenthesis, but it should support both v1 and v0

Comment (heueristik) [reply modules/client-common/src/constants.ts:411]: Ah, in this case, it would be good to write a test for it.

Comment (josemarinas) [reply modules/client-common/src/constants.ts:411]: True, I thought i had tests for both cases, my bad

File: modules/client-common/test/unit/encoding.test.ts (hunk 4 above)
Comment (josemarinas) [reply modules/client-common/test/unit/encoding.test.ts:212]: Mmm, i thing is going to be a bit messier to change it just here and leave in the other places as a harcoded string an just here as a function with, repeat, I can considerchanging it But I'd do it in all places at the same time

**Review by heueristik — APPROVED** [review]

**Pull request merged**